src/main/java/org/reaktivity/nukleus/amqp/internal/stream/AmqpServerFactory.java

```diff
@@ -736,6 +738,14 @@ private int decodePlainFrame(
             final AmqpDescribedType descriptor = performative.kind();
             final AmqpServerDecoder decoder = decodersByPerformative.getOrDefault(descriptor, decodeUnknownType);
             server.decodeChannel = frameHeader.channel();
+
+            AmqpServer.AmqpSession session = server.sessions.get(server.decodeChannel);
+            if (session != null && session.sessionState == AmqpSessionState.DISCARDING && descriptor != END)
+            {
+                progress = limit;
```

Comment:
Shouldn't we make sure to advance only to the end of the frame?
For example, what if a `TRANSFER` and `END` frame would be sent in the same packet and arrive in the same `DATA` payload?